### PR TITLE
bubble_sort

### PR DESCRIPTION
**Pull request by evs062009**

### CODE REVIEW

File: src/BubbleSort.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import java.util.Arrays;
+
+public class BubbleSort {
+
+    public static int [] sort(int [] ints) {
+        int [] newArr = Arrays.copyOf(ints, ints.length);
+        boolean noSwap = true;
+
+        for (int i = newArr.length - 1; i > 0; i--) {
```
Comment (aleksashyn) [inline src/BubbleSort.java]: Try to add more human readable names

File: src/BubbleSort.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+            for (int j = 0; j < i; j++) {
+                if (newArr[j] > newArr[j + 1]){
+                    int temp = newArr[j];
+                    newArr[j] = newArr[j + 1];
+                    newArr[j + 1] = temp;
+                    noSwap = false;
```
Comment (aleksashyn) [inline src/BubbleSort.java:15]: Why you use this variable?

File: src/BubbleSort.java (hunk 1 above)
Comment (evs062009) [reply src/BubbleSort.java]: Do you mean ‘noSwap’ or something else?

Comment (aleksashyn) [reply src/BubbleSort.java]: i, j etc

File: src/BubbleSort.java (hunk 2 above)
Comment (evs062009) [reply src/BubbleSort.java:15]: If there is no swap (no elements change) in inner loop, it means array is already sorted. And we exit the method by ‘breack’.

Comment (evs062009) [reply src/BubbleSort.java:15]: If there was swap (elements change), ‘noSwap’ is set to false. And we continue to sort.

File: src/BubbleSort.java (hunk 1 above)
Comment (evs062009) [reply src/BubbleSort.java]: i, j are used only as counters in loops, and as arrays elements indexes in this loops. I thought it`s normal practice) How do you recommend me to rename such variables (please give me an example)?

Comment (aleksashyn) [reply src/BubbleSort.java]: It's depends on, but in this case i like counter of iterations and j like array index

Comment (aleksashyn) [reply src/BubbleSort.java]: newArr -> numbers

Comment (evs062009) [reply src/BubbleSort.java]: done

**Review by aleksashyn — APPROVED** [review]
Something like that

**Pull request merged**